test/page.spec.js

```diff
@@ -0,0 +1,3812 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const fs = require('fs');
+const path = require('path');
+const {helper} = require('../lib/helper');
+if (process.env.COVERAGE)
+  helper.recordPublicAPICoverage();
+
+const PROJECT_ROOT = fs.existsSync(path.join(__dirname, '..', 'package.json')) ? path.join(__dirname, '..') : path.join(__dirname, '..', '..');
+
+const puppeteer = require(PROJECT_ROOT);
+const DeviceDescriptors = require(path.join(PROJECT_ROOT, 'DeviceDescriptors'));
+const iPhone = DeviceDescriptors['iPhone 6'];
+const iPhoneLandscape = DeviceDescriptors['iPhone 6 landscape'];
+
+const FrameUtils = require('./frame-utils');
+
+const YELLOW_COLOR = '\x1b[33m';
+const RESET_COLOR = '\x1b[0m';
+
+const headless = (process.env.HEADLESS || 'true').trim().toLowerCase() === 'true';
+const slowMo = parseInt((process.env.SLOW_MO || '0').trim(), 10);
+const executablePath = process.env.CHROME;
+
+console.log('Testing on Node', process.version);
```

Comment:
Why do we have this code here? I'd expect it to be only in `test.js`

Reply:
moved to `test.js`. 10x